src/__tests__/scheduling/appointments/new/NewAppointment.test.tsx

```diff
@@ -98,212 +104,148 @@ describe('New Appointment', () => {
-      await act(async () => {
-        wrapper = await setup()
-      })
+      const { container } = setup()
+
       const expectedError = {
         message: 'scheduling.appointment.errors.createAppointmentError',
         startDateTime: 'scheduling.appointment.errors.startDateMustBeBeforeEndDate',
       }
+
       const expectedAppointment = {
-        patient: 'Mr Popo',
+        patient: expectedPatient.fullName,
         startDateTime: new Date(2020, 10, 10, 0, 0, 0, 0).toISOString(),
         endDateTime: new Date(1957, 10, 10, 0, 0, 0, 0).toISOString(),
         location: 'location',
         reason: 'reason',
-        type: 'type',
+        type: 'routine',
       } as Appointment
 
-      act(() => {
-        const appointmentDetailForm = wrapper.find(AppointmentDetailForm)
-        const onFieldChange = appointmentDetailForm.prop('onFieldChange')
-        onFieldChange('patient', expectedAppointment.patient)
-        onFieldChange('startDateTime', expectedAppointment.startDateTime)
-        onFieldChange('endDateTime', expectedAppointment.endDateTime)
+      userEvent.type(
+        screen.getByPlaceholderText(/scheduling\.appointment\.patient/i),
+        expectedAppointment.patient,
+      )
+      fireEvent.change(container.querySelectorAll('.react-datepicker__input-container input')[0], {
+        target: { value: expectedAppointment.startDateTime },
       })
-
-      wrapper.update()
-
-      const saveButton = wrapper.find(Button).at(0)
-      expect(saveButton.text().trim()).toEqual('scheduling.appointments.createAppointment')
-      const onClick = saveButton.prop('onClick') as any
-
-      await act(async () => {
-        await onClick()
+      fireEvent.change(container.querySelectorAll('.react-datepicker__input-container input')[1], {
+        target: { value: expectedAppointment.endDateTime },
       })
-      wrapper.update()
-      const alert = wrapper.find(Alert)
-      const typeahead = wrapper.find(Typeahead)
-      const dateInput = wrapper.find(DateTimePickerWithLabelFormGroup).at(0)
+      userEvent.click(screen.getByText(/scheduling.appointments.createAppointment/i))
 
+      expect(screen.getByText(expectedError.message)).toBeInTheDocument()
+      expect(screen.getByPlaceholderText(/scheduling\.appointment\.patient/i)).toHaveClass(
+        'is-invalid',
+      )
+      expect(container.querySelectorAll('.react-datepicker__input-container input')[0]).toHaveClass(
+        'is-invalid',
+      )
+      expect(screen.getByText(expectedError.startDateTime)).toBeInTheDocument()
       expect(AppointmentRepository.save).toHaveBeenCalledTimes(0)
-      expect(alert.prop('message')).toEqual(expectedError.message)
-      expect(typeahead.prop('isInvalid')).toBeTruthy()
-      expect(dateInput.prop('isInvalid')).toBeTruthy()
-      expect(dateInput.prop('feedback')).toEqual(expectedError.startDateTime)
     })
 
     it('should call AppointmentRepo.save when save button is clicked', async () => {
```

Comment:
I made a change in Dashboard.tsx to test something else. I eventually reverted to code back to what it is currently. Rerunning tests failed 4 tests including this one.

![image](https://user-images.githubusercontent.com/5468546/103903062-0c1edb80-5104-11eb-9a2a-84221f2115dc.png)

Reply:
a rerun of all tests passed so not sure. Perhaps it was just a temporary failure.

Reply:
Same as https://github.com/HospitalRun/hospitalrun-frontend/pull/2516#discussion_r553542885

Reply:
https://github.com/HospitalRun/hospitalrun-frontend/pull/2516#discussion_r554181501